src/Maybe.js

```diff
@@ -1,45 +1,82 @@
-module.exports = Maybe;
+var util = require('./internal/util.js');
 
 function Maybe(x) {
-    if (!(this instanceof Maybe)) {
-        return new Maybe(x);
-    }
+    throw new Error('Maybe cannot be directly instantiated.');
```

Comment:
In Haskell terms `Maybe` is a type whereas `Just` and `Nothing` are data constructors. It doesn't make sense to create a `Maybe` directly as it's a set of data constructors which produce a value of the type: `Maybe = {Just, Nothing}`.

`Maybe.from` provides the old null semantics without making it impossible to represent `Just(null)` and `Just(undefined)`.